Frameworks/Foundation/NSSet.mm

```diff
@@ -554,12 +573,93 @@ - (void)setValue:(id)value forKey:(NSString*)key {
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (NSString*)descriptionWithLocale:(id)locale {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [self descriptionWithLocale:locale indent:0];
+}
+
+/**
+ @Status Interoperable
+*/
+- (NSString*)description {
+    return [self descriptionWithLocale:nil indent:0];
+}
+
+- (NSString*)descriptionWithLocale:(id)locale indent:(NSUInteger)level {
```

Comment:
Well it's worth thinking about that there's a handful of classes that each have a similar API. Like NSOrderedSet, NSDictionary, etc... They're unimplemented at the moment but if each of these produces a similar result then it might be useful to consider. I'm not saying to do it right now, but as we fill out the implementations of Foundation we may have to revisit this.